threads.js

```diff
@@ -731,15 +995,56 @@ ServiceInternal.prototype.onrequest = function(request) {
   }
 };
 
-ServiceInternal.prototype.onmethod = function(method) {
+/**
+ * Called when a client calls
+ * a service's method.
+ *
+ * @param  {Object} method
+ * @return {*}
+ */
+ServicePrivate.prototype.onmethod = function(method) {
   debug('method', method.name);
   var fn = this.methods[method.name];
   if (!fn) throw new Error(ERRORS[4]);
   this.checkMethodCall(method);
-  return fn.apply(this.external, method.args);
+  return fn.apply(this.public, method.args);
 };
 
-ServiceInternal.prototype.respond = function(request, result) {
+/**
+ * Called during `client.stream()`
+ */
+ServicePrivate.prototype.onstream = function(method, request) {
+  debug('stream', method.name);
+  var fn = this.streams[method.name];
+  if (!fn) throw new Error(ERRORS[4]);
+
+  var id = method.id;
+  var client = request.client;
+  var channel = this.channels[client];
+  var stream = new ServiceStream(id, channel, this.id, client);
```

Comment:
Number of args passsed to `ServiceStream` is getting lengthy. Perhaps a `params` object might be more readable. What do you think?